### DISCUSSION TITLE
hsluv vs hsv

### DISCUSSION THREAD

**Discussion by crimesp** [rydmike/flex_color_picker#89] [Ideas]
Hi mike I'm a big fan of your work thank you so much for filling the 'material gap'  as well as some awesome toolage!

 I've gone down the rabbit hole of trying to pick the perfect colours full my app,  which uses three  different 'primary'  colours depending on the context it's in.  I may have a future app with eight of these primary colors. 
 
 Anyway - I discovered HSLUV  and I think I will use  this to ensure my colours have the same perceived luminosity,  or potentially in some cases deliberate increased luminosity.
 
  have you ever looked at this kind of color methodolgy and  can you think of any reasons not to use colours all at the same perceived  luminance?  I'm  planning on forking color picker to use HSLUV  just so I can get the right values I need easily.
  
   is this a feature that you think you would like in flex color  picker,  as an alternate mode or alternate picker?  if so - I'll be a little bit more careful with my approach
  
   I'll be combining the  code from this project: 
https://github.com/bernaferrari/hsluv-dart

 thanks, paul
 
 example From the above project -  it's an animated gift on the project  page,  the top picker is HSV the bottom one is the HSLUV
 
![image](https://github.com/rydmike/flex_color_picker/assets/1911935/57a4dd6c-c7eb-4b52-b8d0-4cd3bac87b2a)

 there is also a live flutter example here  with some interesting features that may be of interest to you - https://bernaferrari.github.io/hsluv-dart/#/

**Comment by rydmike** [comment]
Hi @crimesp and thanks for asking. And yes I am familiar with HSLUV, although I actually think that Google's HCT color space is even better for even color gradients (The original article behind it https://material.io/blog/science-of-color-design), but a matter of taste of course.

I do plan to add a color picker where you can select the used color space and use sliders for its props to select a color. I thought I had HSLUV listed as an option to maybe include as well, but apparently not. I will add it to the list.

Related issue: https://github.com/rydmike/flex_color_picker/issues/19

I do hope I can get to working on and improving the ColorPicker later this year and release a totally revised and more flexible and advanced version.

**Comment by crimesp** [comment]
Hi there thank you so much for the response.  hadn't realized that there was HCT under the hood with all the tonal things. Will use this instead of HSLUv

 I've managed to achieve my aims  by picking the colors I wanted and just running them through a tone filter  to create swatch to give me some balanced material colors.
 
  I  tried picking the colors from the toned versions but I  got better results just using the material defaults  & Generating standard tones  to create  some material colors
  
  
  For reference for any readers who want to do the same I've attached my ugly code).
  
  
  
[material_palette_generator.dart.txt](https://github.com/user-attachments/files/16041930/material_palette_generator.dart.txt)
